### PR TITLE
release/public-v1: use NCEP libraries and ESMF from NCEPLIBS umbrella build

### PR DESCRIPTION
**Pull request by climbfuji**
For the two regression testing platforms, use the NCEP libraries and ESMF from the NCEPLIBS umbrella build instead of the operational ones.

This requires a new baseline on Cheyenne because of a netCDF version upgrade (the netCDF version is written to the file's HDF5 metadata, silly). Otherwise, results are bit-for-bit identical for both Intel and GNU (even though the Intel compiler is updated from 19.0.2 to 19.0.5). On hera, the existing baseline is correct, but needs to be copied/synced/linked to the new date tag 20200224.

Also adding a cmake config for stampede so that the ufs-weather-model's `build.sh` script can be used (config is identical to hera). This config has been tested.

Also updating the submodule pointer for stochastic_physics following the recent merge (documentation only, rst files) @pjpegion.

### CODE REVIEW

File: cmake/configure_stampede.intel.cmake [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+option(OPENMP  "Enable OpenMP threading" ON)
+option(AVX2    "Enable AVX2 instruction set" ON)
+
+option(INLINE_POST "Enable inline post" OFF)
+
+include( cmake/${CMAKE_Fortran_COMPILER_ID}.cmake )
+
+set(NEMSIO_INC $ENV{NEMSIO_INC})
+set(POST_INC $ENV{POST_INC})
+set(NCEP_LIBS $ENV{POST_LIB} $ENV{NEMSIO_LIB} $ENV{G2_LIB4} $ENV{G2TMPL_LIB} $ENV{BACIO_LIB4} $ENV{SP_LIBd} $ENV{W3EMC_LIBd} $ENV{W3NCO_LIBd} $ENV{CRTM_LIB} $ENV{PNG_LIB} $ENV{JASPER_LIB} $ENV{Z_LIB})
```
Comment (junwang-noaa) [inline cmake/configure_stampede.intel.cmake]: Do we need post_lib, crtm, png, jasper libs, those are for inline post/ww3, which public release does not support, also I see the NCEPlibs/1.0.0beta02 on hera seem not having them.

Comment (climbfuji) [reply cmake/configure_stampede.intel.cmake]: Good point, I will remove those. Only from the new stampede config? Or also from hera and cheyenne when using the NCEPLIBS umbrella build?

**Comment by DusanJovic-NOAA** [comment]
I copied ufs-public-release-20200219 to ufs-public-release-20200224 on Hera.

**Comment by climbfuji** [comment]
Thanks! Will run the tests in about an hour or two ...

File: cmake/configure_stampede.intel.cmake (hunk 1 above)
Comment (junwang-noaa) [reply cmake/configure_stampede.intel.cmake]: I prefer to clean it up in all the config files when using the NCEPLIBS umbrella build.

Comment (climbfuji) [reply cmake/configure_stampede.intel.cmake]: Ok, will do!

Comment (climbfuji) [reply cmake/configure_stampede.intel.cmake]: I removed those and other libraries not used (g2tmpl, g2).

**Comment by climbfuji** [comment]
Results on hera are b4b identical after making these changes, regression test logs have been updated. Tests on Cheyenne are still running against the newly created baseline.

[rt_hera_intel.log](https://github.com/ufs-community/ufs-weather-model/files/4247174/rt_hera_intel.log)

**Comment by climbfuji** [comment]
Regression tests successful on Cheyenne+GNU, logs have been updated. Intel to come soon.

[rt_cheyenne_gnu_create.log](https://github.com/ufs-community/ufs-weather-model/files/4247364/rt_cheyenne_gnu_create.log)
[rt_cheyenne_gnu.log](https://github.com/ufs-community/ufs-weather-model/files/4247365/rt_cheyenne_gnu.log)

**Comment by climbfuji** [comment]
Regression tests successful on Cheyenne+Intel. The logs are attached here and need to be updated tomorrow before merging the PR or in a follow-up PR as last time.

[RegressionTests_cheyenne.intel.log](https://github.com/ufs-community/ufs-weather-model/files/4247837/RegressionTests_cheyenne.intel.log)
[Compile_cheyenne.intel.log](https://github.com/ufs-community/ufs-weather-model/files/4247836/Compile_cheyenne.intel.log)

[rt_cheyenne_intel.log](https://github.com/ufs-community/ufs-weather-model/files/4247838/rt_cheyenne_intel.log)
[rt_cheyenne_intel_create.log](https://github.com/ufs-community/ufs-weather-model/files/4247835/rt_cheyenne_intel_create.log)

**Comment by climbfuji** [comment]
@DusanJovic-NOAA @junwang-noaa Please merge this at your convenience, all reg test log files have been updated. Thanks!

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]

**Pull request merged**